sonar-db/src/main/java/org/sonar/db/qualityprofile/QProfileChangeDto.java

```diff
@@ -17,98 +17,101 @@
+public class QProfileChangeDto {
 
   private String key;
   private String profileKey;
-  private String message;
-  private String type;
-  private String action;
-  private String author;
+  // can't be named "type" because it's a reserved word in Oracle
+  // (used by Mybatis to map DB column with DTO field)
+  private String changeType;
+  private String login;
   private String data;
-  private Date createdAt;
-
-  public ActivityDto setKey(String key) {
-    this.key = key;
-    return this;
-  }
+  private long createdAt;
 
   public String getKey() {
     return key;
   }
 
+  public QProfileChangeDto setKey(String s) {
+    this.key = s;
+    return this;
+  }
+
   public String getProfileKey() {
     return profileKey;
   }
 
-  public ActivityDto setProfileKey(String profileKey) {
-    this.profileKey = profileKey;
+  public QProfileChangeDto setProfileKey(String s) {
+    this.profileKey = s;
     return this;
   }
 
-  public Date getCreatedAt() {
-    return createdAt;
+  public String getChangeType() {
+    return changeType;
   }
 
-  public ActivityDto setCreatedAt(Date createdAt) {
-    this.createdAt = createdAt;
+  public QProfileChangeDto setChangeType(String s) {
+    this.changeType = s;
     return this;
   }
 
-  public String getType() {
-    return type;
+  public long getCreatedAt() {
+    return createdAt;
   }
 
-  public ActivityDto setType(String type) {
-    this.type = type;
+  public QProfileChangeDto setCreatedAt(long l) {
+    this.createdAt = l;
     return this;
   }
 
-  public String getAuthor() {
-    return author;
+  @CheckForNull
+  public String getLogin() {
+    return login;
   }
 
-  public ActivityDto setAuthor(@Nullable String author) {
-    this.author = author;
+  public QProfileChangeDto setLogin(@Nullable String s) {
+    this.login = s;
     return this;
   }
 
+  @CheckForNull
   public String getData() {
     return data;
   }
 
-  public ActivityDto setData(String data) {
-    this.data = data;
-    return this;
-  }
-
-  public String getMessage() {
-    return message;
+  public Map<String, String> getDataAsMap() {
+    if (data == null) {
+      return Collections.emptyMap();
+    }
+    return KeyValueFormat.parse(data);
   }
 
-  public ActivityDto setMessage(@Nullable String message) {
-    this.message = message;
+  public QProfileChangeDto setData(@Nullable String csv) {
+    this.data = csv;
     return this;
   }
 
-  public String getAction() {
-    return action;
-  }
-
-  public ActivityDto setAction(String action) {
-    this.action = action;
+  public QProfileChangeDto setData(@Nullable Map m) {
+    if (m == null || m.isEmpty()) {
+      this.data = null;
+    } else {
+      this.data = KeyValueFormat.format(m);
+    }
     return this;
   }
 
   @Override
   public String toString() {
-    return ReflectionToStringBuilder.toString(this, ToStringStyle.MULTI_LINE_STYLE);
+    return ReflectionToStringBuilder.toString(this, ToStringStyle.SHORT_PREFIX_STYLE);
```

Comment:
using reflection in `toString()` is a non sense, imho
reflection is costly and no one want's a dumb `toString()` to have perf impacts, especially when method generators exist in IDEs.
Changing this DTOs could be a good opportunity to get rid of that debt